### ISSUE TITLE
libpcap 1.10.0 breaks ipv6toolkit (relative to 1.8.1)

### ISSUE THREAD

**Issue by mirabilos** [the-tcpdump-group/libpcap#1119]
Not sure whether the issue is with libpcap, ipv6toolkit or both, but I think it might be best to bring it to both sides’ attention (especially given I kinda need a network testing tool at the moment).

Upgrading from Debian buster, which ships libpcap 1.8.1, to bullseye, which ships 1.10.0, breaks ipv6toolkit’s neighbour discovery: https://github.com/fgont/ipv6toolkit/issues/78 and https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1016129

**Comment by guyharris** [comment]
I commented on fgont/ipv6toolkit#78, noting that, by default, ipv6toolkit's error messages amount to "Some unspecified bad thing happened while doing X", and asking that the tools be run with `-vv` which, I'm guessing from a quick check of the code, force them to report errors from iibpcap calls, to try to figure out whether a particular call is failing and, if so, what error it reports when failing.

**Comment by infrastation** [comment]
Thank you for the bug report. Please note the test systems are different in other regards besides just libpcap (including the kernel), so the issue needs to be distilled to one particular environment and a list of steps to reproduce. Debian bug report seems to be a good place to do that. Ideally it would be great to use git bisection to tell the exact change that has introduced the difference in libpcap behaviour (which might or might not be a bug). When you have a proof the issue you observe is specific to libpcap, please post an update here.

**Comment by mirabilos** [comment]
Denis Ovsienko dixit:

>Thank you for the bug report. Please note the test systems are
>different in other regards besides just libpcap (including the kernel),

In the other bugreport, I wrote I tracked it down to just libpcap
by installing the older distro’s libpcap in the newer distro (fixes
it) viceque versa (breaks it).

Kernel was the second thing I tested, after the application itself.

bye,
//mirabilos
-- 
Sometimes they [people] care too much: pretty printers [and syntax highligh-
ting, d.A.] mechanically produce pretty output that accentuates irrelevant
detail in the program, which is as sensible as putting all the prepositions
in English text in bold font.	-- Rob Pike in "Notes on Programming in C"
